src/app/list/list.component.ts

```diff
@@ -0,0 +1,107 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { ListService } from '../list-service/list-service';
+import { Category } from 'src/app/classes/category';
+import { Task } from 'src/app/classes/task';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { ViewChild } from '@angular/core';
+import { FormGroupDirective } from '@angular/forms';
+import { CdkDragDrop, moveItemInArray } from '@angular/cdk/drag-drop';
+import { Key } from '../constants/keyboard-keys';
+
+@Component({
+  selector: 'app-list',
+  templateUrl: './list.component.html',
+  styleUrls: ['./list.component.css'],
+})
+export class ListComponent implements OnInit {
+  @Input() category!: Category;
+
+  @ViewChild(FormGroupDirective, { static: false })
+  formDirective!: FormGroupDirective;
+
+  requriedFormControl = new FormControl('', [Validators.required]);
+  tasks: Task[] = [];
+  newTaskName: string = '';
+  showError: boolean = true;
+
+  listForm: FormGroup;
+
+  constructor(public listService: ListService) {
+    this.listForm = new FormGroup({});
+  }
+  addTask() {
+    if (this.requriedFormControl.errors) {
```

Comment:
```suggestion
  constructor(public listService: ListService) {
    this.listForm = new FormGroup({});
  }
  
  addTask() {
    if (this.requriedFormControl.errors) {
```

Make sure there is a newline between function definitions